### PR TITLE
[JENKINS-62371] add the ability to have an icon in the section header

### PR DESCRIPTION
**Pull request by jtnord**
See [JENKINS-62371](https://issues.jenkins-ci.org/browse/JENKINS-62371).

### Proposed changelog entries

* Developer: add the ability to add an icon to a form section header
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/resources/lib/form/section.jelly [hunk 1]
```diff
@@ -34,13 +34,19 @@ THE SOFTWARE.
...
+    </st:attribute>
   </st:documentation>
 
   <st:adjunct includes="lib.form.section_" />
   <f:rowSet name="${attrs.name}">
     <j:if test="${attrs.title!=null}">
       <f:block>
         <div class="section-header">
+          <j:if test="${attrs.icon!=null}">
+            <img src="${icon}"/>
```
Comment (fqueiruga) [inline core/src/main/resources/lib/form/section.jelly:48]: What would be a way to support SVG icons through `<svg>` tags and not only as img srcs?. Also, we should provide an option to add alt text for the section header, defaulting to empty string `alt=""`.

**Review by fqueiruga — CHANGES_REQUESTED** [review]

**Review by oleg-nenashev — COMMENTED** [review]
Would be nice to see a reference implementation for it

File: core/src/main/resources/lib/form/section.jelly (hunk 1 above)
Comment (jtnord) [reply core/src/main/resources/lib/form/section.jelly:48]: so the alt should be "" and not configurable imo.

this is for a quick visual display only and polluting screen readers etc with alt text is a bad design.

I thought about just adding a flag and allowing raw html in the title (so you can do whatever) but thought that would be abusable and lead to a potentialy wiered looking UI.  same for SVG, these are small icons..  YAGNI

**Comment by jtnord** [comment]
> Would be nice to see a reference implementation for it

was going to do one to hi examples plugin.  Wanted to get quick feedback on is this generally considered a good or bad thing to begin with.

File: core/src/main/resources/lib/form/section.jelly (hunk 1 above)
Comment (fqueiruga) [reply core/src/main/resources/lib/form/section.jelly:48]: Sounds reasonable 👍 . You are right about the YAGNI. We can start with this and if there's a need for SVG sprite icons here we can just add a way for it later.

Comment (fqueiruga) [reply core/src/main/resources/lib/form/section.jelly:48]: Feel free to resolve this one

Comment (jtnord) [reply core/src/main/resources/lib/form/section.jelly:48]: I guess we should have a discussion about the hight/width attributes for scaling though?

Comment (fqueiruga) [reply core/src/main/resources/lib/form/section.jelly:48]: I would suggest setting it relative to the font size. Maybe you could try with a 1.1 or 1.2 em width and height.

**Comment by daniel-beck** [comment]
> provides more recognisable context when a section is about GitHub or some well known external service that has a well known icon

This justification makes some sense, but it would probably be good if the documentation for this included some non-technical guidance to developers around terms of use and such of these icons.

OTOH we've always had some trouble around inconsistencies in icons used in Jenkins, for example in the sidepanels and Manage Jenkins. When there are no inconsistencies, AFAICT it would be cumbersome to get rid of the Tango icons and replace them with something else due to the number and independence of components involved. So do we really want to add support for more locations for icons?

I also wonder whether it would be expected for a well-maintained Mailer plugin to add an "envelope" icon, i.e. is this intended to be limited to "special" icons (e.g. integration with other services) or is it supposed to be done everywhere it can be?

**Comment by jtnord** [comment]
SVG icons are now more pervasive in Jenkins. Did not generate the discussion I intended to spark and is now has merge conflict so just closing.  Can be re opened if needed

**Pull request closed without merging**